projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts

```diff
@@ -1971,35 +1974,34 @@ export class IgxTimePickerComponent implements
      * @hidden
      */
     public onInput(event): void {
-        const val = event.target.value;
+        const inputMask = event.target.value;
         const oldVal = new Date(this.value);
 
-        this.isNotEmpty = val !== this.parseMask(false);
+        this.isNotEmpty = inputMask !== this.parseMask(false);
 
         // handle cases where all empty positions (promts) are filled and we want to update
         // timepicker own value property if it is a valid Date
-        if (val.indexOf(this.promptChar) === -1) {
-            if (this._isEntryValid(val)) {
-                const newVal = this.convertMinMaxValue(val);
+        if (inputMask.indexOf(this.promptChar) === -1) {
+            if (this._isEntryValid(inputMask)) {
+                const newVal = this.convertMinMaxValue(inputMask);
                 if (oldVal.getTime() !== newVal.getTime()) {
                     this.value = newVal;
                 }
             } else {
                 const args: IgxTimePickerValidationFailedEventArgs = {
                     timePicker: this,
-                    currentValue: val,
+                    currentValue: inputMask,
                     setThroughUI: false
                 };
                 this.onValidationFailed.emit(args);
             }
             // handle cases where the user deletes the display value (when pressing backspace or delete)
-        } else if (!this.value || !val || val === this.parseMask(false)) {
+        } else if (!this.value || !inputMask || inputMask === this.parseMask(false)) {
             this.isNotEmpty = false;
-
-            this.value.setHours(0, 0);
-            this.displayValue = val;
-
-            if (oldVal.getTime() !== this.value.getTime()) {
+            // TODO: refactoring - this.value should be null #6585
+            this.value?.setHours(0, 0, 0);
+            this.displayValue = inputMask;
+            if (oldVal.getTime() !== this.value?.getTime()) {
```

Comment:
in case there is no initial value, and you delete the entire mask the component will fire `onValueChanged` which is wrong. This is better in this PR as on master this throws an exception. Please add a comment for future refactoring to stop emit this event in this case.